### ISSUE TITLE
Java 11 Target, Hibernate and JaCoCo not working

### ISSUE THREAD

**Issue by pschyma** [quarkusio/quarkus#6800]
**Describe the bug**
Updating to Quarkus 1.2.0 in a project that uses Hibernate (Panache) and JaCoCo and setting the Java target version to 11 breaks the tests. It seems like the byte code augmentation is interfering.

**Expected behavior**
Tests are executed and a coverage report is created.

**Actual behavior**
The maven build is aborted with an error:
```
❯ mvnc verify
[INFO] Scanning for projects...
[INFO] 
[INFO] --------------< org.acme:tests-with-coverage-quickstart >---------------
[INFO] Building tests-with-coverage-quickstart 1.0-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- maven-clean-plugin:2.5:clean (default-clean) @ tests-with-coverage-quickstart ---
[INFO] Deleting /data/Code/quarkus-java-11-panache-jacoco/target
[INFO] 
[INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ tests-with-coverage-quickstart ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Copying 1 resource
[INFO] 
[INFO] --- maven-compiler-plugin:3.1:compile (default-compile) @ tests-with-coverage-quickstart ---
[INFO] Changes detected - recompiling the module!
[INFO] Compiling 3 source files to /data/Code/quarkus-java-11-panache-jacoco/target/classes
[INFO] 
[INFO] --- jacoco-maven-plugin:0.8.5:instrument (instrument-ut) @ tests-with-coverage-quickstart ---
[INFO] 
[INFO] --- maven-resources-plugin:2.6:testResources (default-testResources) @ tests-with-coverage-quickstart ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Copying 1 resource
[INFO] 
[INFO] --- maven-compiler-plugin:3.1:testCompile (default-testCompile) @ tests-with-coverage-quickstart ---
[INFO] Changes detected - recompiling the module!
[INFO] Compiling 2 source files to /data/Code/quarkus-java-11-panache-jacoco/target/test-classes
[INFO] 
[INFO] --- maven-surefire-plugin:2.22.1:test (default-test) @ tests-with-coverage-quickstart ---
[INFO] 
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running org.acme.quickstart.GreetingServiceTest
[INFO] H2 database started in TCP server mode; server status: TCP server running at tcp://192.168.110.169:9092 (only local connections)
11:00:51,586 INFO  [org.jbo.threads] JBoss Threads version 3.0.0.Final
11:00:52,065 INFO  [org.hib.jpa.boo.int.PersistenceXmlParser] HHH000318: Could not find any META-INF/persistence.xml file in the classpath
11:00:52,320 INFO  [org.hib.Version] HHH000412: Hibernate Core {5.4.10.Final}
[INFO] H2 database was shut down; server status: Not started
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 3.913 s <<< FAILURE! - in org.acme.quickstart.GreetingServiceTest
[ERROR] testGreetingService  Time elapsed: 0.008 s  <<< ERROR!
org.junit.jupiter.api.extension.TestInstantiationException: TestInstanceFactory [io.quarkus.test.junit.QuarkusTestExtension] failed to instantiate test class [org.acme.quickstart.GreetingServiceTest]: Failed to enhance class org.acme.quickstart.GreetingEntity
Caused by: org.hibernate.bytecode.enhance.spi.EnhancementException: Failed to enhance class org.acme.quickstart.GreetingEntity
Caused by: java.lang.UnsupportedOperationException: This feature requires ASM7

[INFO] 
[INFO] Results:
[INFO] 
[ERROR] Errors: 
[ERROR]   GreetingServiceTest.testGreetingService » TestInstantiation TestInstanceFactor...
[INFO] 
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  9.960 s
[INFO] Finished at: 2020-01-27T11:00:53+01:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.22.1:test (default-test) on project tests-with-coverage-quickstart: There are test failures.
[ERROR] 
[ERROR] Please refer to /data/Code/quarkus-java-11-panache-jacoco/target/surefire-reports for the individual test results.
[ERROR] Please refer to dump files (if any exist) [date].dump, [date]-jvmRun[N].dump and [date].dumpstream.
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
```

**To Reproduce**
1. clone the https://github.com/pschyma/quarkus-java-11-panache-jacoco repository (the code is based on the jacoco starter project with a simple hibernate entity)
2. execute `./mvnw clean verify`

**Environment (please complete the following information):**
 - Output of `uname -a` or `ver`: 
`Linux voyager 5.4.15-arch1-1 #1 SMP PREEMPT Sun, 26 Jan 2020 09:48:50 +0000 x86_64 GNU/Linux`
 - Output of `java -version`: 
```
openjdk version "11.0.6" 2020-01-14
OpenJDK Runtime Environment GraalVM CE 19.3.1 (build 11.0.6+9-jvmci-19.3-b07)
OpenJDK 64-Bit Server VM GraalVM CE 19.3.1 (build 11.0.6+9-jvmci-19.3-b07, mixed mode, sharing)
```
 - Quarkus version or git rev: 1.2.0

**Additional context**
`./mvnw clean quarkus:dev` works without any issues.

**Comment by gwenneg** [comment]
Hi @pschyma and thanks for reporting this!

This issue is caused by the ASM version (ASM5) used in [PersistentAttributeTransformer](https://github.com/hibernate/hibernate-orm/blob/master/hibernate-core/src/main/java/org/hibernate/bytecode/enhance/internal/bytebuddy/PersistentAttributeTransformer.java#L137) which is part of Hibernate. The ASM version used in Quarkus expects `ASM7` instead of `ASM5` during the `GreetingEntity` entity enhancement.

**Comment by Sanne** [comment]
@gwenneg 's fix was merged in Hibernate ORM / master, but this will take an ORM release and upgrade Quarkus to fix here.

https://hibernate.atlassian.net/browse/HHH-13833

**Comment by gwenneg** [comment]
Thanks for your help with this @Sanne!

**Comment by gsmet** [comment]
@Sanne let's plan for a new ORM release before the CR1 so before February 21st. I suppose we will have other fixes but this alone will allow us to support JDK 14 (probably) so that's a nice win.

**Comment by Sanne** [comment]
Sure

**Comment by pschyma** [comment]
Thanks for the investigation and update.

**Comment by Sanne** [comment]
fixed by #7083
